org.eclipse.scanning.api/src/org/eclipse/scanning/api/scan/IFilePathService.java

```diff
@@ -21,14 +21,23 @@
 public interface IFilePathService {
 
 	/**
-	 * Determine and return the next file path to write to.
+	 * Determine and return the next file path to write to. The file will be in the default output directory.
+	 * @param template for instance the sample name, may be null or empty string if no template is required.
 	 * @return next file path
-	 * @param template, for instance the sample name, may be null or empty string if no template is required.
 	 * @throws Exception if the next path cannot be calculated for any reason
 	 */
 	String getNextPath(String template) throws Exception;
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Define and throw a dedicated exception instead of using a generic one. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS00112)